node/pkg/guardiansigner/guardiansigner.go

```diff
@@ -0,0 +1,66 @@
+package guardiansigner
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"strings"
+)
+
+// The types of guardian signers that are supported
+type SignerType int
+
+const (
+	InvalidSignerType SignerType = iota
+	FileSignerType
+)
+
+// GuardianSigner interface
+type GuardianSigner interface {
+	// TODO: document that the signer implementations assume they are recieving a keccack256 hash
+	Sign(hash []byte) (sig []byte, err error)
+	PublicKey() (pubKey ecdsa.PublicKey)
+	Verify(sig []byte, hash []byte) (valid bool, err error)
+}
+
+func NewGuardianSignerFromUri(signerUri string, unsafeDevMode bool) (GuardianSigner, error) {
+
+	// Get the signer type
+	signerType, signerKeyConfig := ParseSignerUri(signerUri)
+
+	switch signerType {
+	case FileSignerType:
+		return NewFileSigner(unsafeDevMode, signerKeyConfig)
+	case InvalidSignerType:
+		return nil, fmt.Errorf("unsupported guardian signer type")
```

Comment:
This case seems identical to the `default` one. Is it explicitly needed?

Reply:
Yeah I'm dropping the `InvalidSignerType` case. Wanted it in there to make use of it, but I really think it's a waste of code.